app/models/publisher.rb

```diff
@@ -323,8 +332,37 @@ def paypal_locale?(locale)
     locale == 'ja'
   end
 
+  def brave_payable?
+    paypal_connection&.verified_account || uphold_connection&.payable?
+  end
+
+  def country
+    provider_country = uphold_connection&.country || paypal_connection&.country
+
+    provider_country.to_s.upcase
+  end
+
+  def valid_promo_country?
+    PromoRegistration::RESTRICTED_COUNTRIES.exclude?(country)
+  end
+
+  def promo_registrable?
+    # If the user doesn't have the referral_kyc_flag on then we can register them still.
+    return true unless referral_kyc_required?
+
+    # Otherwise they must be brave payable and from a valid country
+    (brave_payable? && valid_promo_country?) || (promo_expiration_time.present? && promo_not_expired?)
```

Comment:
This shoudl only contain one check otherwise promo expiration time won't hold

Reply:
https://github.com/brave-intl/publishers/pull/2733/commits/99a33d783c7fcaa136639550816a0fcc9bd39936